docs/modules/ROOT/pages/references/style-guide.adoc

```diff
@@ -0,0 +1,273 @@
+= Style Guide
+
+The Jsonnet related parts of this guide build on top of https://github.com/databricks/jsonnet-style-guide/blob/30d4d54444c1001ba9561afd7c8de405630a2ca7/README.md[Jsonnet Guide] by https://databricks.com/[Databricks].
+
+http://creativecommons.org/licenses/by-nc-sa/4.0/[image:https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png[Creative Commons License]] +
+This work is licensed under a http://creativecommons.org/licenses/by-nc-sa/4.0/[Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License].
+
+== General Style
+
+* Encode text files with UTF-8.
+* Use Unix line endings (lf).
+* Always end a file with a single empty line.
+  This doesn't apply to empty files (for example `.gitkeep`).
+* Use two spaces for indentation except for `Makefile` where a single tab is to be used.
+* Trim whitespaces from the end of a line.
+
+[NOTE]
+====
+Repositories of Project Syn provide an https://editorconfig.org/[EditorConfig].
+When using a https://editorconfig.org/#download[supported editor], space types and number of spaces will automatically be done correctly.
+====
+
+== Asciidoc Style
+
+* Don't split senctences across lines but keep them on one single line.
+* Start a new line for each and every sentence.
+* Follow the https://docs.microsoft.com/en-us/style-guide/welcome/[Microsoft Writing Style Guide].
```

Comment:
According to the Microsoft Writing Style Guide, section headings should be sentence case and not title case, which puts this style guide in violation of itself (cf. https://docs.microsoft.com/en-us/style-guide/capitalization#sentence-style-capitalization-in-titles-and-headings).

Reply:
So much for: the style gets automatically tested.